AdyenNetworking/APIClient/Errors.swift

```diff
@@ -0,0 +1,47 @@
+//
+//  APIError.swift
+//  AdyenNetworking
+//
+//  Created by Mohamed Eldoheiri on 7/27/21.
+//
+
+import Foundation
+
+/// Describes possible `APIClient` errors
+public enum APIClientError: LocalizedError {
+    
+    case invalidResponse
+    
+    var errorDescription: String {
+        return "Invalid Response"
+    }
+    
+}
+
+/// Represents an HTTP error object.
+public struct HTTPError<E: ErrorResponse>: LocalizedError {
```

Comment:
I think HTTPError is an `HTTResponse` with an `ErrorResponse`, I feel like `HTTPError` as a separate type is redundant.

Reply:
that's a good idea. How would we model that? 
for example we could `typealias HTTPErrorResponse<E: ErrorResponse> = HTTPResponse<E>`, but I want HTTPErrorResponse to be throwable, so it needs to conform to error as well

Reply:
if we make HTTPResponse a class, we end up with this. This makes `HTTPErrorResponse` throwable.
```swift
public class HTTPResponse<R: Response> { ... }

public final class HTTPErrorResponse<E: ErrorResponse>: HTTPResponse<E>, Error { /* nothing here */ }
```

Reply:
I think `typealias HTTPErrorResponse<E: ErrorResponse> = HTTPResponse<E>` is a good idea.

but I don't understand the need for `HTTPErrorResponse` to be throwable. but anyway if you really need `HTTPErrorResponse` to be an error, maybe you can do this:

```Swift
extension HTTPResponse : Error where R : Error { }
```

Reply:
I think it's better to have it throwable, so it's always clear if the request was successful.